core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -261,14 +269,19 @@ private[spark] class ExecutorAllocationManager(
   private def updateAndSyncNumExecutorsTarget(now: Long): Int = synchronized {
     val maxNeeded = maxNumExecutorsNeeded
 
-    if (maxNeeded < numExecutorsTarget) {
+    if (initializing && (maxNeeded < numExecutorsTarget)) {
+      // when ExecutorAllocationManager is still in initializing status,
+      // and the max number of needed executors is less than the initial number of executors,
+      // we just return without doing anything.
```

Comment:
I would rephrase this:
```
// Do not cancel requests while we are still initializing,
// otherwise the first job may have to ramp up unnecessarily
```